### ISSUE TITLE
ERROR:use of undeclared identifier 'XC_FAMILY_HYB_GGA'

### ISSUE THREAD

**Issue by Luc1anoo** [deepmodeling/abacus-develop#2782]
### Describe the bug

These errors occured during compiling "Libxc"

### To Reproduce

**cmake:**
(base) root@bohrium-17819-1028724:/abacus-develop# cmake -B build -DENABLE_LIBRI=ON -DENABLE_LIBXC=ON -DLibxc_DIR=/Libxc -DLIBRI_DIR=/abacus-develop/deps/LibRI -DLIBCOMM_DIR=/abacus-develop/deps/LibComm
-- Found git: attempting to get commit info...
-- Current commit hash: b3c50b650
-- Last commit date: Fri Jul 28 17:41:41 2023 +0800
CMake Warning at CMakeLists.txt:394 (message):
  RI related features are only stable if compiled with Intel compiler.

-- Found Libxc: version 7.0.0
-- Configuring done
-- Generating done
-- Build files have been written to: /abacus-develop/build

**make:**
(base) root@bohrium-17819-1028724:/abacus-develop# cd build
(base) root@bohrium-17819-1028724:/abacus-develop/build# make -j2 LIBXC_DIR=/Libxc

**errors during compiling:**  
...
[ 33%] Built target elecstate
Consolidate compiler generated dependencies of target xc_
[ 33%] Building CXX object source/module_hamilt_general/module_xc/CMakeFiles/xc_.dir/xc_functional.cpp.o
[ 33%] Building CXX object source/module_hamilt_general/module_surchem/CMakeFiles/surchem.dir/cal_epsilon.cpp.o
[ 33%] Building CXX object source/module_hamilt_general/module_xc/CMakeFiles/xc_.dir/xc_functional_vxc.cpp.o
[ 34%] Building CXX object source/module_hamilt_general/module_surchem/CMakeFiles/surchem.dir/cal_pseudo.cpp.o
/abacus-develop/source/module_hamilt_general/module_xc/xc_functional_vxc.cpp:213:22: error: use of undeclared identifier 'XC_FAMILY_HYB_GGA'
                case XC_FAMILY_HYB_GGA:
                     ^
/abacus-develop/source/module_hamilt_general/module_xc/xc_functional_vxc.cpp:338:18: error: use of undeclared identifier 'XC_FAMILY_HYB_GGA'
            case XC_FAMILY_HYB_GGA:
                 ^
/abacus-develop/source/module_hamilt_general/module_xc/xc_functional_vxc.cpp:369:71: error: use of undeclared identifier 'XC_FAMILY_HYB_GGA'
        if(func.info->family == XC_FAMILY_GGA || func.info->family == XC_FAMILY_HYB_GGA)
                                                                      ^
3 errors generated.
make[2]: *** [source/module_hamilt_general/module_xc/CMakeFiles/xc_.dir/build.make:90: source/module_hamilt_general/module_xc/CMakeFiles/xc_.dir/xc_functional_vxc.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1159: source/module_hamilt_general/module_xc/CMakeFiles/xc_.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 34%] Building CXX object source/module_hamilt_general/module_surchem/CMakeFiles/surchem.dir/cal_totn.cpp.o
[ 34%] Building CXX object source/module_hamilt_general/module_surchem/CMakeFiles/surchem.dir/cal_vcav.cpp.o
[ 34%] Building CXX object source/module_hamilt_general/module_surchem/CMakeFiles/surchem.dir/cal_vel.cpp.o
[ 35%] Building CXX object source/module_hamilt_general/module_surchem/CMakeFiles/surchem.dir/corrected_energy.cpp.o
[ 35%] Building CXX object source/module_hamilt_general/module_surchem/CMakeFiles/surchem.dir/minimize_cg.cpp.o
[ 35%] Building CXX object source/module_hamilt_general/module_surchem/CMakeFiles/surchem.dir/sol_force.cpp.o
[ 35%] Built target surchem
make: *** [Makefile:136: all] Error 2

### Environment

OS:Ubuntu 22.04.1 LTS
Compler:gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0
ABACUS: v3.3.1

**Comment by dyzheng** [comment]
http://abacus.deepmodeling.com/en/latest/advanced/install.html#build-with-libxc It is mentioned version must larger than 5.1.7, but not mentioned the upper limit version, maybe you can try some old version.

**Comment by caic99** [comment]
Hi @Luc1anoo ,
Please make sure you've installed the correct libxc version. Actually I can not find libxc 7 anywhere.

**Comment by PeizeLin** [comment]
You can download libxc from [official website](https://www.tddft.org/programs/libxc/download/).

**Comment by hongriTianqi** [comment]
@Luc1anoo Please see if the above suggestions solve your issue. Please let us know if you still have problems. Thanks!

**Comment by Luc1anoo** [comment]
The problem is solved. It turns out  that there is a directory named "Libxc" with the path "/usr/share/cmake/Libxc" and it has CMakeList.txt file. I deleted the Libxc file and finally the cmake command can find my 6.2.2 version Libxc with the path I chose in the command "-DLibxc_DIR=/Libxc". At the end I still don't know how could it find a version 7.0.0 Libxc ,but the bug is gone anyway. Thanks for the  help!
